src/legacy/ui/field_formats/field_format.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { isFunction, transform, size, cloneDeep, get, defaults } from 'lodash';
+import { createCustomFieldFormat } from './converters/custom';
+import { ContentType, FieldFormatConvert, IFieldFormat, FieldFormatConvertFunction } from './types';
+
+import { htmlContentTypeSetup, textContentTypeSetup } from './content_types';
+
+const DEFAULT_CONTEXT_TYPE = 'text';
+
+export type FieldFormatConvert = { [key: string]: Function } | FieldFormatConvertFunction;
+
+export abstract class FieldFormat implements IFieldFormat {
```

Comment:
nit: Would it work to instead export `IFieldFormat` like this?
```ts
export type IFieldFormat = PublicMethodsOf<FieldFormat>;
```

We have a type for this in the root `/typings` directory... might help remove some of the duplication in the `types.ts`. Plus then we could put the comments for each method inline here, which would help with readability.

Reply:
np, fixed